pandas/core/generic.py

```diff
@@ -8508,6 +8508,58 @@ def align(
         -------
         (left, right) : ({klass}, type of other)
             Aligned objects.
+
+        Examples
+        --------
+        >>> left = pd.DataFrame(
+        ...     [[1, 2, 3, 4], [6, 7, 8, 9]], columns=["D", "B", "E", "A"], index=[1, 2]
+        ... )
+
+        >>> right = pd.DataFrame(
+        ...     [[10, 20, 30, 40], [60, 70, 80, 90], [600, 700, 800, 900]],
+        ...     columns=["A", "B", "C", "D"],
+        ...     index=[2, 3, 4],
+        ... )
+
+        >>> left
+           D  B  E  A
+        1  1  2  3  4
+        2  6  7  8  9
+
+        >>> right
+            A    B    C    D
+        2   10   20   30   40
+        3   60   70   80   90
+        4  600  700  800  900
+
+        >>> a1, a2 = left.align(right, join="outer", axis=1)
```

Comment:
last thing - for consistency with the arguments, can we name the original dataframes `df` and `other`, and then the results `left` and `right`? So
```suggestion
        >>> df = pd.DataFrame(
        ...     [[1, 2, 3, 4], [6, 7, 8, 9]], columns=["D", "B", "E", "A"], index=[1, 2]
        ... )

        >>> other = pd.DataFrame(
        ...     [[10, 20, 30, 40], [60, 70, 80, 90], [600, 700, 800, 900]],
        ...     columns=["A", "B", "C", "D"],
        ...     index=[2, 3, 4],
        ... )

        >>> df
           D  B  E  A
        1  1  2  3  4
        2  6  7  8  9

        >>> other
            A    B    C    D
        2   10   20   30   40
        3   60   70   80   90
        4  600  700  800  900

        >>> left, right = left.align(right, join="outer", axis=1)
```

(likewise for the axis=0 example)